devito/types/relational.py

```diff
@@ -0,0 +1,260 @@
+"""User API to specify relationals."""
+
+import sympy
+
+__all__ = ['Le', 'Lt', 'Ge', 'Gt', 'Ne']
+
+
+class Le(sympy.Le):
+    """
+    A less-than or equal ("<=") relation between two objects, the left-hand side and the
+    right-hand side. Can be used to build conditionals but not directly to
+    construct an Operator.
+
+    The left-hand side may be a Function or a SparseFunction. The right-hand
+    side may be any arbitrary expressions with numbers, Dimensions, Constants,
+    Functions and SparseFunctions as operands.
+
+    Parameters
+    ----------
+    lhs : Function or SparseFunction
+        The left-hand side.
+    rhs : expr-like, optional
+        The right-hand side. Defaults to 0.
+    subdomain : SubDomain, optional
+        To restrict the evalaution of the relation to a particular sub-region in the
+        computational domain.
+
+    Examples
+    --------
+    Le may be used to express a relation (e.g. in a Subdomain).
+
+    >>> from devito import Grid, Function, ConditionalDimension, Eq, Operator
+    >>> from devito.types import Le
+    >>> grid = Grid(shape=(8, 8))
+    >>> g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
+    >>> x, y = grid.dimensions
+    >>> Le(g, 1)
+    g(x, y) <= 1
+    """
+
+    def __new__(cls, lhs, rhs=0, subdomain=None, **kwargs):
+        kwargs.update({'evaluate': False})
+        obj = sympy.Le.__new__(cls, lhs, rhs, **kwargs)
+        obj._subdomain = subdomain
+        return obj
+
+    @property
+    def subdomain(self):
+        """The SubDomain in which the Le is defined."""
+        return self._subdomain
+
+    @property
+    def negated(self):
+        return ops.get(self.func)(*self.args)
```

Comment:
what is `ops` ? :O

Reply:
edit: ah I see, at the end...

unnecessary I'd say?

why not just

```
return Gt(*self.args)
```

?

Reply:
Generallity, and matches sympy Relational

Reply:
I think it'd be cleaner to just return `Gt(...)` but OK

Reply:
Can that be marked as resolved?